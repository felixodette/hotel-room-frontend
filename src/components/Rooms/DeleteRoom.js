import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRooms, deleteRoom } from '../../redux/rooms';
import '../styles/DeleteRoom.css';

const DeleteRoom = () => {
  const dispatch = useDispatch();
  const rooms = useSelector((state) => state.rooms.rooms);
  const [message, setMessage] = useState('');

  const checkResponse = (response, id) => {
    if (response === 204) {
      dispatch(deleteRoom(id));
    } else {
      setMessage(<div className="add-room-error-notification border mt-1 py-1 px-2 bg-dark text-danger rounded p-1">ERROR: Reserved rooms can`t be deleted</div>);
    }
  };
  setInterval(() => { setMessage(''); }, 5000);

  const deleteRoomHandler = (id) => {
    const requestOptions = {
      method: 'DELETE',
      headers: { 'Content-type': 'application/json; charset=UTF-8' },
    };

    fetch(`http://localhost:3000/api/v1/rooms/${id}`, requestOptions)
      .then((response) => checkResponse(response.status, id));
  };

  useEffect(() => {
    dispatch(getRooms());
  }, [dispatch]);
  return (
    <div id="delete-room-container" className="container-fluid d-flex flex-column align-items-center h-100 mb-5">
      <h2 id="add-room-heading" className="text-center  mt-5 fw-bold fs-1 text-white text-uppercase">Delete Room</h2>
      <hr id="add-room-hr" />
      <p className="text-center  fs-6 text-white">
        Would you like to delete your hotel room from our website?
        {' '}
        <br />
        You can do it with one click!
      </p>
      <div>
        {message}
      </div>
      <div id="delete-room-table" className="table w-50 table-hover">
        <table className="table w-100 table-hover">
          <thead>
            <tr>
              <th scope="col" className="text-white">Room Name</th>
              <th scope="col" className="text-center text-white">Action</th>
            </tr>
          </thead>
          <tbody>
            {rooms.map((room) => (
              <tr className="room" key={room.id}>
                <td><p className="text-white mt-3" id="room-name">{room.name}</p></td>
                <td className="text-center">
                  <button
                    id="delete-room-btn"
                    className="mt-2"
                    type="button"
                    onClick={() => deleteRoomHandler(room.id)}
                  >
                    Delete
                  </button>
                </td>
              </tr>
            ))}
          </tbody>

        </table>
      </div>
    </div>
  );
};

export default DeleteRoom;
