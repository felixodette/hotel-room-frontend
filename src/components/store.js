import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../redux/reducers/index';

const initialState = {};
const middleware = [thunk];
const store = createStore(
  rootReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
/* eslint-disable */
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()));
/* eslint-enable */

export default store;
