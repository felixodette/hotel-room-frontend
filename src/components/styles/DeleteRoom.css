#delete-room-container {
  min-height: 100%;
  background: linear-gradient(0deg, rgba(151, 191, 17, 0.8), rgba(151, 191, 17, 0.8)), url("https://cache.marriott.com/content/dam/marriott-renditions/BJVEB/bjveb-deluxe-0003-hor-wide.jpg?output-quality=70&interpolation=progressive-bilinear&downsize=1336px");
  background-size: cover;
  height: 100vh;
}

#delete-room-btn {
  border: 0;
  width: 100px;
  height: 40px;
  border-radius: 20px;
  color: #97bf11;
  transition-duration: 0.4s;

}

#delete-room-btn:hover {
  background-color: transparent;
  border: 1px solid white;
  color: white;
}

@media (max-width: 768px) {
#delete-room-table {
  width: 100% !important;
}
}

