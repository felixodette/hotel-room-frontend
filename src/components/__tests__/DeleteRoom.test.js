import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import DeleteRoom from '../Rooms/DeleteRoom';
import store from '../../redux/configureStore';

describe('DeleteRoom', () => {
  render(
    <Provider store={store}>
      <DeleteRoom />
    </Provider>,
  );

  test('Test the dom for success rendering elements', () => {
    const title = screen.getByText(/Action/);
    expect(title).toBeInTheDocument();
  });

  test('Test the dom for non existing element on the page', () => {
    const title = screen.queryByText('example');
    expect(title).not.toBeInTheDocument();
  });

  test('It renders the Rooms component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <DeleteRoom />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('It renders the Rooms component correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DeleteRoom />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
