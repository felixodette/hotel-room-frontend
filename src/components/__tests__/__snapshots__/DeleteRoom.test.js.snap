// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteRoom It renders the Rooms component correctly 1`] = `
<div>
  <div
    class="container-fluid d-flex flex-column align-items-center h-100 mb-5"
    id="delete-room-container"
  >
    <h2
      class="text-center  mt-5 fw-bold fs-1 text-white text-uppercase"
      id="add-room-heading"
    >
      Delete Room
    </h2>
    <hr
      id="add-room-hr"
    />
    <p
      class="text-center  fs-6 text-white"
    >
      Would you like to delete your hotel room from our website?
       
      <br />
      You can do it with one click!
    </p>
    <div />
    <table
      class="table w-50 table-hover"
      id="delete-room-table"
    >
      <thead>
        <tr>
          <th
            class="text-white"
            scope="col"
          >
            Room Name
          </th>
          <th
            class="text-center text-white"
            scope="col"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`DeleteRoom It renders the Rooms component correctly 2`] = `
<div
  className="container-fluid d-flex flex-column align-items-center h-100 mb-5"
  id="delete-room-container"
>
  <h2
    className="text-center  mt-5 fw-bold fs-1 text-white text-uppercase"
    id="add-room-heading"
  >
    Delete Room
  </h2>
  <hr
    id="add-room-hr"
  />
  <p
    className="text-center  fs-6 text-white"
  >
    Would you like to delete your hotel room from our website?
     
    <br />
    You can do it with one click!
  </p>
  <div />
  <table
    className="table w-50 table-hover"
    id="delete-room-table"
  >
    <thead>
      <tr>
        <th
          className="text-white"
          scope="col"
        >
          Room Name
        </th>
        <th
          className="text-center text-white"
          scope="col"
        >
          Action
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
